update-format-crau/src/delta_update.rs

```diff
@@ -147,27 +147,42 @@ pub fn get_data_blobs<'a>(f: &'a mut BufReader<File>, header: &'a DeltaUpdateFil
 // parse_signature_data takes bytes slices for signature and digest of data blobs,
 // and path to public key, to parse and verify the signature.
 // Return only actual signature data, without version and special fields.
-pub fn parse_signature_data(sigbytes: &[u8], digest: &[u8], pubkeyfile: &str) -> Option<Box<[u8]>> {
+pub fn parse_signature_data(sigbytes: &[u8], digest: &[u8], pubkeyfile: &str) -> Result<Box<[u8]>> {
     // Signatures has a container of the fields, i.e. version, data, and
     // special fields.
     let sigmessage = match proto::Signatures::parse_from_bytes(sigbytes) {
         Ok(data) => data,
-        _ => return None,
+        _ => bail!("failed to parse signature messages"),
     };
 
     // sigmessages.signatures[] has a single element in case of dev update payloads,
     // while it could have multiple elements in case of production update payloads.
     // For now we assume only dev update payloads are supported.
     // Return the first valid signature, iterate into the next slot if invalid.
-    sigmessage.signatures.iter()
-        .find_map(|sig|
-            verify_sig_pubkey(digest, sig, pubkeyfile)
-            .map(Vec::into_boxed_slice))
+    let mut sigres: Option<Vec<u8>> = None;
+
+    for sig in sigmessage.signatures {
+        match verify_sig_pubkey(digest, &sig, pubkeyfile) {
+            Ok(sbox) => {
+                sigres = Some(sbox.to_vec());
+                break
+            }
+            _ => {
+                info!("failed to verify signature, jumping to the next slot");
+                continue
+            }
+        };
+    }
+
+    match sigres {
+        Some(s) => Ok(s.into()),
+        None => bail!("failed to find a valid signature in any slot"),
+    }
```

Comment:
```suggestion
    bail!("failed to find a valid signature in any slot"),
```

Reply:
I am not sure I understand.
The above change makes the function always fail.
Just double checked with my local test, and it fails for sure.

Reply:
Done